### PR TITLE
Add component metadata YAML for bindings.redis.

### PR DESCRIPTION
**Pull request by tmacam**
# Description

Add component metadata YAML for bindings.redis.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: bindings/redis/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+      Allows specifying a default Time-to-live (TTL) in seconds that will be
+      applied to every state store request unless TTL is explicitly defined
+      via the request metadata.
+    example: "600"
+    type: number
+  - name: queryIndexes
+    required: false
+    description: Indexing schemas for querying JSON objects
+    example: "see Querying JSON objects"
+    type: string
```
Comment (tmacam) [inline bindings/redis/metadata.yaml:207]: If I comment all this out and run `make check-component-metadata && make bundle-component-metadata` the command only complains about the following properties:

```
The following components are missing metadata in their metadata.yaml:
{
  "bindings/redis": {
    "maxretries": "int",
    "maxretrybackoff": "time.Duration",
    "queryindexes": "string",
    "ttlinseconds": "*int"
  }
}
```

Like 🤯. What is happening here?

Comment (DeepanshuA) [reply bindings/redis/metadata.yaml:207]: But, these are optional attributes, as far as I understand. Am I missing something?

Comment (DeepanshuA) [reply bindings/redis/metadata.yaml:207]: But, all these mentioned 4 metadata are optional. Am I missing something?

Comment (tmacam) [reply bindings/redis/metadata.yaml:207]: If I only define those 4 metadata and nothing else that is in this metadata.yaml file, `make check-component-data` would pass.

What is puzzling me is that these fields (or a structure hosting only these fields) are not defined in any of the structs used for metadata and settings by bindings/redis. 

> Am I missing something?

That is exactly my question. I am even wondering: am I missing something really obvious and should I instead define only these 4 fields `make check-component-data`  is requiring instead ?

Comment (tmacam) [reply bindings/redis/metadata.yaml]: ALrigh! We found the issue and it is addressed on #2803.

Pending its merge before procedding. Moving this back to draft.

**Review by berndverst — CHANGES_REQUESTED** [review]
```
{
  "bindings/redis": {
    "maxlenapprox": "int64",
    "redismaxretries": "int",
    "redismaxretryinterval": "redis.Duration"
  }
}
```

**Review by berndverst — APPROVED** [review]

**Pull request merged**